src/test/scala/com/payalabs/scalajs/react/bridge/ReactBridgeComponentTest.scala

```diff
@@ -5,15 +5,17 @@ import japgolly.scalajs.react.test.ReactTestUtils
 import org.scalatest.FunSuite
 
 import scala.scalajs.js
-
+class NameType (val name :String) extends AnyVal with ReactBridgeObject {
+  override def toJS  = name.asInstanceOf[js.Dynamic]
+}
 class ReactBridgeComponentTest extends FunSuite {
 
   test("scalar properties") {
     case class TestComponent(id: js.UndefOr[String] = js.undefined, className: js.UndefOr[String] = js.undefined,
                              ref: js.UndefOr[String] = js.undefined, key: js.UndefOr[Any] = js.undefined,
-                             name: js.UndefOr[String] = js.undefined, age: js.UndefOr[Int] = js.undefined) extends ReactBridgeComponent
+                             name: js.UndefOr[NameType] = js.undefined, age: js.UndefOr[Int] = js.undefined) extends ReactBridgeComponent
 
-    val testComponent: ReactElement = TestComponent(name = "foo", age = 25)()
+    val testComponent: ReactElement = TestComponent(name = new NameType("foo"), age = 25)()
```

Comment:
Can you move this to a separate test (say "object properties") and make assertions on it? Please take a look at test-component.js. You will need to come up with a string encoding for test purpose. Perhaps tack on "Map" when converting a `Map` and "Object" for object.

p.s. I will remove the print statement (it is a left-over) or you can remove it when you ammend this PR.